### PR TITLE
[#138] Autoreleasing

### LINKED ISSUE #138: Make a new release (preferrably automatically)
# Clarification and motivation

We will inevitably create a new release at some point and we often don't create issues for that, but this time there are two things to notice:
1. Our latest release contains `tezos-client` (a patched version of it) which is not necessary and doesn't work with latest protocols. We shouldn't include it anymore.
2. It'd be nice to create releases automatically, so that we don't have to attach artifacts manually downloaded from CI. I suppose it should be relatively easy to do, we are doing it already in several projects. We can use the official GitLab CLI probably.

# Acceptance criteria

1. A new release is created without `tezos-client` in it.
2. [Optional, but desired] This release is created automatically by CI and auto-creation can be reused later.

### PR DESCRIPTION
**Pull request by rvem**
## Description
We already have a project that has autoreleasing and it's quite
convenient when we want to have a release with some set of assets, so
we're going to use autoreleasing in tzbtc as well.

## Related issue(s)

Resolves #138

#### Related changes (conditional)

- Tests (see [short guidelines](../blob/master/CONTRIBUTING.md#tests))
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - I checked whether I should update the docs and did so if necessary:
    - [x] [README](../blob/master/README.md)
    - [x] Haddock
    - [x] [docs/](../blob/master/docs/)
  - [x] I updated [changelog](../blob/master/ChangeLog.md) unless I am sure my changes are
        really minor.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../blob/master/docs/code-style.md).

### CODE REVIEW

File: ci.nix [hunk 1]
```diff
@@ -41,4 +41,16 @@ rec {
     '';
   contract-doc-dev = contract-doc-with-commit null;
   contract-doc-release = { sha, date }@commitInfo: contract-doc-with-commit commitInfo;
+  gh = (pkgs.callPackage "${pkgs.path}/pkgs/applications/version-management/git-and-tools/gh" {
```
Comment (gromakovsky) [inline ci.nix:44]: What's going on here, is there no `gh` in `nixpkgs` that can be easily used like `reuse` and other tools we use?

**Comment by gromakovsky** [comment]
Looks like something went wrong:
> assets/tzbtc-client-deb/*.deb: no such file or directory

File: ci.nix (hunk 1 above)
Comment (rvem) [reply ci.nix:44]: gh is in the nixpkgs, but the version presented in nixpkgs is old and doesn't support releases stuff, so we have to rebuild the newer version ourselves

**Comment by rvem** [comment]
Fixed

File: ci.nix (hunk 1 above)
Comment (gromakovsky) [reply ci.nix:44]: Okay, maybe let's write a comment about that?

**Review by gromakovsky — APPROVED** [review]

**Pull request merged**